lib/src/group_changes.dart

```diff
@@ -0,0 +1,307 @@
+          after.add(hashIndex[change["commit_hash"]]);
+        }
+        configs.sort();
+        final formattedSet = configs.join(",<br>");
+        configSets[formattedSet] = configs;
+        final summaryData =
+            ByConfigSetData(formattedSet, test, results, before, after);
+        byBlamelist
+            .putIfAbsent(summaryData.after.max,
+                () => Map<int, Map<String, List<ByConfigSetData>>>())
+            .putIfAbsent(summaryData.before.min,
+                () => Map<String, List<ByConfigSetData>>())
+            .putIfAbsent(
+                summaryData.formattedConfigSet, () => List<ByConfigSetData>())
+            .add(summaryData);
+      }
+    }
+  }
+  return byBlamelist;
+}
+
+String prelude() => '''
+<!DOCTYPE html><html><head><title>Results Feed</title>
+<style>
+  span.indent   {padding-left: 4em;}
+  textarea      {font-family: monospace;}
+  span.commit   {font-family: monospace; color: gray;}
+  td            {font-family: monospace;}
+  div.blamelist {font-family: monospace; background-color: lightgray; padding: 10px;}
+  h1            {color: blue;}
+  td            {vertical-align: top;}
+  td.outer      {padding: 10px;}
+  td.nopad      {padding: 0px;}
+  td.nomatch    {background-color: Salmon;}
+  td.match      {background-color: PaleGreen;}
+</style>
+<script>function showBlamelist(id) {
+  if (document.getElementById(id + "-off").style.display == "none") {
+    document.getElementById(id + "-on").style.display = "none";
+    document.getElementById(id + "-off").style.display = "block";
+  } else {
+    document.getElementById(id + "-off").style.display = "none";
+    document.getElementById(id + "-on").style.display = "block";
+  }
+}
+
+function showInline(id) {
+  if (document.getElementById(id + "-off").style.display == "none") {
+    document.getElementById(id + "-on").style.display = "none";
+    document.getElementById(id + "-off").style.display = "inline";
+  } else {
+    document.getElementById(id + "-off").style.display = "none";
+    document.getElementById(id + "-on").style.display = "inline";
+  }
+}
+</script>
+</head><body><h1>Results Feed</h1>
+''';
+
+String postlude() => '''</body></html>''';
+
+String htmlPage(Map<int, Map<int, Map<String, List<ByConfigSetData>>>> data,
+    List<String> hashes, List<String> commitData, List<String> reviewLinks) {
+  StringBuffer page = StringBuffer(prelude());
+
+  page.write("<table>");
+  var after = MinMax();
+  data.keys.forEach(after.add);
+  for (var afterKey = 0; afterKey <= after.max; ++afterKey) {
+    // Info from git about this commit:
+    page.write("<span class='commit'><a href='${reviewLinks[afterKey]}'>"
+        "${hashes[afterKey].substring(0, 8)}</a>&nbsp;&nbsp;"
+        "${commitData[afterKey]}</span><br>");
+    if (!data.containsKey(afterKey)) continue;
+    final beforeKeys = data[afterKey].keys.toList()..sort();
+    for (var beforeKey in beforeKeys) {
```

Comment:
Okay so, we've found a commit that's the earliest seen with the test changes, and now you iterate all the blamelists that end in that commit, starting with the latest commit seen without the test change.